core/modules/parser/QSMySqlListener.cc

```diff
@@ -53,8 +53,25 @@ namespace {
 
 LOG_LOGGER _log = LOG_GET("lsst.qserv.QSMySqlListener");
 
+std::string getQueryString(antlr4::ParserRuleContext * ctx) {
+    return ctx->getStart()->getInputStream()->getText(
+           antlr4::misc::Interval(ctx->getStart()->getStartIndex(), ctx->getStop()->getStopIndex()));
 }
 
+template <typename T>
+std::string getTypeName() {
+    int status;
+    return abi::__cxa_demangle(typeid(T).name(),0,0,&status);
```

Comment:
Is this really cross-compiler portable? My understanding is that this is a specific feature of the GCC family of compilers.

Reply:
My understanding is that it's not totally portable. It is *very* useful. I encapsulated it so that if we encounter problems with other compilers it can be worked around. As far as I know there's not a universal solution to get the class name, ya?

Reply:
Agreed on all counts.